subsys/bluetooth/host/hci_core.c

```diff
@@ -4686,6 +4659,36 @@ int bt_addr_le_create_static(bt_addr_le_t *addr)
 	return 0;
 }
 
+static u8_t bt_read_public_addr(bt_addr_le_t *addr)
```

Comment:
Why u8_t? That seems like a strange choice. The usual `int`-style error reporting would seem appropriate here.

Reply:
It matches bt_read_static_addr, returning the number of addresses read. there can be multiple static addresses, but only one public one. So in truth this could be a bool, I'm assigning bt_dev.id_count to the return value of this function.